src/components/LCircle.vue

```diff
@@ -14,12 +19,15 @@ export default {
     const leafletRef = ref({});
     const ready = ref(false);
 
+    const useGlobalLeaflet = inject(GLOBAL_LEAFLET_OPT);
     const addLayer = inject("addLayer");
 
     const { options, methods } = circleSetup(props, leafletRef, context);
 
     onMounted(async () => {
-      const { circle, DomEvent } = await import("leaflet/dist/leaflet-src.esm");
+      const { circle, DomEvent } = useGlobalLeaflet
+        ? WINDOW_OR_GLOBAL.L
```

Comment:
@mikeu Did we check if like this we break tree-shaking? I have a felling we do, we should maybe import this dynamically

Reply:
@DonNicoJs that is a good question that I haven't checked into yet, no.

Are you worried that this change will break tree shaking in general, even when `useGlobalLeaflet` is _not_ true? If so, then that is definitely an issue that will need further investigation.

I am not sure why you're worried about right here in particular though. Running `const { circle } = L` shouldn't really have anything to do with tree shaking, should it? It will simply execute as expected if that `L` already exists, or throw an error if it doesn't. The code here is basically the same as what [already exists in master](https://github.com/vue-leaflet/vue-leaflet/blob/8433a8986ac8c95ca091f4eaaadace8438cfc047/src/components/LMap.vue#L229-L231) in the LMap component, just without the conditional asynchronous import if the global variable isn't set.

What your question has now made me somewhat more worried about though is the [behaviour in LMap](https://github.com/vue-leaflet/vue-leaflet/pull/125/files/2b33787ab8cf2f0b1c29149017bde7833188190d#diff-c062e5b98deef6d3ca2604206fe64f4ac3a74887e252285b8559ee1e08e1b932R225-R227). Perhaps the fact that it has that `await import("leaflet")` even conditionally is enough to break tree shaking, which may very well not be able to identify whether `WINDOW_OR_GLOBAL.L` will be defined, and so might _always_ include the whole Leaflet package in the bundle...

If that's the case, then we should remove those lines and simply make it the responsibility of the user to guarantee that `L` will be globally defined before the map mounts if they choose to set `useGlobalLeaflet=true`, perhaps with a friendly error message if they haven't.

Do you happen to already have an environment set up to quickly test the results of tree shaking? Also, we've been speaking of "tree shaking" as a single unified feature that we intend to support, but is that a fair statement or do different algorithms mean we have to explicitly know whether the library works with tree shaking under Webpack, Rollup, esbuild, ...?